api/middleware.go

```diff
@@ -0,0 +1,112 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+// ManagerURLMiddleware injects X-Forwarded-* headers based on manager-url setting.
+// This middleware reads the manager-url setting and, when configured, overrides
+// the X-Forwarded-Proto, X-Forwarded-Host, and X-Forwarded-Port headers to ensure
+// API responses (actions and links fields) contain the correct external URL instead
+// of internal pod IPs.
+func ManagerURLMiddleware(s *Server) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			managerURL, err := s.m.GetSettingValueExisted(types.SettingNameManagerURL)
+			if err != nil {
+				logrus.WithError(err).Warn("Failed to get manager-url setting, using default behavior")
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			if managerURL == "" {
+				// Setting not configured, use default behavior
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			if err := validateManagerURL(managerURL); err != nil {
+				logrus.WithError(err).Warnf("Invalid manager-url setting value: %s", managerURL)
+				next.ServeHTTP(w, r)
+				return
+			}
```

Comment:
Can move this part to the [Settings CR validator](https://github.com/longhorn/longhorn-manager/blob/master/types/setting.go#L2397) so that the invalid URL will be rejected by the webhook.

Reply:
Done